include/eosio/vm/backend.hpp

```diff
@@ -321,7 +322,13 @@ namespace eosio { namespace vm {
                _timed_out.store(true, std::memory_order_release);
                mod->allocator.disable_code();
             });
-            std::forward<F>(f)();
+
+            using return_type = std::invoke_result_t<F>;
+            if constexpr (std::is_void_v<return_type>) {
```

Comment:
I'm suspicious this check isn't actually required.

Reply:
yeah I tried and replacing all this with just `return std::forward<F>(f)();` compiled okay (on spring's return_status branch)

Reply:
Thanks! You are correct. It is now even cleaner.